src/Cake.Common/Tools/VSTest/VSTestRunner.cs

```diff
@@ -18,8 +18,10 @@ namespace Cake.Common.Tools.VSTest
     /// </summary>
     public sealed class VSTestRunner : Tool<VSTestSettings>
     {
+        private const string VSTestConsoleExecutableName = "vstest.console.exe";
```

Comment:
Do you mean like this? `protected override IEnumerable<string> GetToolExecutableNames() => new[] { "vstest.console.exe" };`

And `Tool<VSTestSettings> will resolve it automatically on `GetAlternativeToolPaths`?

Reply:
Yes like that.

As you can see [here](https://github.com/cake-build/cake/blob/develop/src/Cake.Core/Tooling/Tool.cs#L284) in `Tool<TSettings>` class, when `Run(..)` is called, it will eventually hit that method which will start the tool resolution. By returning the correct executable name(s) from `GetToolExecutableNames()`, tool locator will try to resolve the tool using the default conventions, which is defiend [here](https://cakebuild.net/docs/tools/tool-resolution). The last step in that convention is `GetAlternativeToolPaths()`, but if the `exe` can be found in `tools\` directory, then that method won't even be called.

@cake-build/cake-team do you happen to know why the correct executable name haven't been registered in this tool?

Reply:
Fixed!